app/models/track.rb

```diff
@@ -43,6 +86,70 @@ def to_param
     short_name
   end
 
+  def transition_possible?(transition)
+    self.class.state_machine.events_for(current_state).include?(transition)
+  end
+
+  # Gives the role of the track_organizer to the submitter
+  def assign_role_to_submitter
+    submitter.add_role 'track_organizer', self
+  end
+
+  # Revokes the track organizer role and removes the track from events that have it set
+  def revoke_role_and_cleanup
+    role = Role.find_by(name: 'track_organizer', resource: self)
+
+    if role
+      role.users.each do |user|
+        user.remove_role 'track_organizer', self
+      end
+    end
+
+    events.each do |event|
+      event.track = nil
```

Comment:
Because the track organizer can't accept and schedule events yet, I'm only removing the track from the event. When the scheduling for tracks is implemented, I will also destroy the track's schedule and revert the event back to the new state